engine/schema/src/main/java/com/cloud/configuration/dao/ResourceCountDaoImpl.java

```diff
@@ -126,8 +126,10 @@ public void updateDomainCount(long domainId, ResourceType type, boolean incremen
         delta = increment ? delta : delta * -1;
 
         ResourceCountVO resourceCountVO = findByOwnerAndType(domainId, ResourceOwnerType.Domain, type);
-        resourceCountVO.setCount(resourceCountVO.getCount() + delta);
-        update(resourceCountVO.getId(), resourceCountVO);
+        if (resourceCountVO != null) {
```

Comment:
If you do not find a resource to update, should not you break the execution flow?
It seems that if the `resourceCountVO` is null, then we must have something wrong in the database.

Reply:
updateDomainCount is an @Deprecated method, can we ignore it?

Reply:
I did not check the code. Is it used somewhere else?
I think we can ignore (it has been working so far...).

Reply:
Then, what about if you delete it?

Reply:
hum, you mean delete the updateDomainCount ?

Reply:
Yes. If it is not used. So, it is better to delete it.